src/lib/components/TokenImg.tsx

```diff
@@ -21,18 +21,21 @@ function TokenImg({ token, ...rest }: TokenImgProps) {
   const srcs = useCurrencyLogoURIs(tokenInfo)
 
   const [attempt, setAttempt] = useState(0)
-  const onError = useCallback((e) => {
-    if (e.target.src) badSrcs.add(e.target.src)
-    setAttempt((attempt) => ++attempt)
-  }, [])
-
   const src = useMemo(() => {
     // Trigger a re-render when an error occurs.
     void attempt
 
     return srcs.find((src) => !badSrcs.has(src))
   }, [attempt, srcs])
-
+  const onError = useCallback(
+    (e) => {
+      if (src) badSrcs.add(src)
+      setAttempt((attempt) => ++attempt)
```

Comment:
any particular reason you do this as a mutation of the local scope before the setter returns the new value?

Reply:
`attempt` needs to be updated to force `src` to update, because `badSrcs` is external to React's state. This is just incrementing `attempt` to ensure it is updated.